src/general/world_effects/patch_events/GlobalGlaciationEvent.cs

```diff
@@ -0,0 +1,258 @@
+        foreach (var configuration in IceChunksConfigurations)
+        {
+            var iceChunkConfiguration = templateBiome.Conditions.Chunks[configuration];
+            iceChunkConfiguration.Density *= 10;
+            patch.Biome.Chunks.Add(Prefix + configuration, iceChunkConfiguration);
+        }
+    }
+
+    private void LogEvent(Patch patch, double totalTimePassed)
+    {
+        patch.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT"),
+            true, true, "GlobalGlaciationEvent.svg");
+
+        if (patch.Visibility == MapElementVisibility.Shown)
+        {
+            targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_EVENT_LOG", patch.Name),
+                true, false, "GlobalGlaciationEvent.svg");
+        }
+
+        patch.AddPatchEventRecord(WorldEffectVisuals.GlobalGlaciation, totalTimePassed);
+    }
+
+    private void LogBeginningOfGlaciation()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_START_EVENT_LOG"),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void LogEndOfGlaciation()
+    {
+        targetWorld.LogEvent(new LocalizedString("GLOBAL_GLACIATION_END_EVENT_LOG"),
+            true, true, "GlobalGlaciationEvent.svg");
+    }
+
+    private void FinishEvent()
+    {
+        hasEventAlreadyHappened = true;
+        foreach (var index in modifiedPatchesIds)
+        {
+            if (!targetWorld.Map.Patches.TryGetValue(index, out var patch))
+                continue;
+
+            PatchSnapshot patchSnapshot = patch.History[eventDuration];
+
+            ResetBackground(patch, patchSnapshot);
+            ResetEnvironment(patch, patchSnapshot);
+            RemoveChunks(patch);
+        }
+
+        LogEndOfGlaciation();
+    }
+
+    private void ResetBackground(Patch patch, PatchSnapshot patchSnapshot)
+    {
+        var biomeBackground = patchSnapshot.Background;
+        var biomeLightColour = patchSnapshot.LightColour;
+        patch.BiomeTemplate.Background = biomeBackground;
+        patch.BiomeTemplate.Sunlight.Colour = biomeLightColour;
+        patch.CurrentSnapshot.Background = biomeBackground;
+        patch.CurrentSnapshot.LightColour = biomeLightColour;
+    }
+
+    private void ResetEnvironment(Patch patch, PatchSnapshot patchSnapshot)
+    {
+        var currentTemperature = patch.Biome.ChangeableCompounds[Compound.Temperature];
+        var currentSunlight = patch.Biome.ChangeableCompounds[Compound.Sunlight];
+        var previousTemperature = patchSnapshot.Biome.ChangeableCompounds[Compound.Temperature];
+        var previousSunlight = patchSnapshot.Biome.ChangeableCompounds[Compound.Sunlight];
+
+        currentTemperature.Ambient = previousTemperature.Ambient - currentTemperature.Ambient;
+        currentSunlight.Ambient = previousSunlight.Ambient - currentSunlight.Ambient;
+
+        var changes = new Dictionary<Compound, float>
```

Comment:
I think it would be much easier to use the `ModifyLongTermCondition(Compound compound, BiomeCompoundProperties newValue)` overload here to just apply the old values back.